src/crypto/sign.js

```diff
@@ -114,22 +133,15 @@ export function verifyMessageWithTwoPublicKeys(signedMessage, publicKey, secondP
 		throw new Error('Invalid second publicKey, expected 32-byte publicKey');
 	}
 
-	const secondSignatureVerified = naclInstance.crypto_sign_open(
-		signedMessageBytes, secondPublicKeyBytes,
+	const firstSignatureVerified = naclInstance.crypto_sign_verify_detached(
+		signatureBytes, messageBytes, publicKeyBytes,
 	);
 
-	if (!secondSignatureVerified) {
-		throw new Error('Invalid signature second publicKey, cannot verify message');
-	}
-
-	const firstSignatureVerified = naclInstance.crypto_sign_open(
-		secondSignatureVerified, publicKeyBytes,
+	const secondSignatureVerified = naclInstance.crypto_sign_verify_detached(
+		secondSignatureBytes, messageBytes, secondPublicKeyBytes,
 	);
 
-	if (!firstSignatureVerified) {
-		throw new Error('Invalid signature first publicKey, cannot verify message');
-	}
-	return naclInstance.decode_utf8(firstSignatureVerified);
+	return firstSignatureVerified && secondSignatureVerified;
```

Comment:
Not sure how noticeable it would be, but more performant would be to have these as functions so that you don't have to wait for both verifications if the first one fails. I.e.:
```js
const verifyFirstSignature = () => naclInstance.crypto_sign_verify_detached(signatureBytes, messageBytes, publicKeyBytes);
const verifySecondSignature = () => naclInstance.crypto_sign_verify_detached(secondSignatureBytes, messageBytes, secondPublicKeyBytes);
return verifyFirstSignature() && verifySecondSignature();